src/cmd-cloud-prune

```diff
@@ -0,0 +1,291 @@
+    # The Python bindings don't support passing a config file.
+    # The alternative is to manually pass ACCESS_KEY and SECRET_KEY which isn't favourable.
+    if args.aws_config_file:
+        os.environ["AWS_CONFIG_FILE"] = args.aws_config_file
+    s3_client = boto3.client("s3")
+
+    # These lists are up to date as of schema hash
+    # 4c19aed3b3d84af278780bff63728510bb3e70613e4c4eef8cabd7939eb31bd8. If changing
+    # this hash, ensure that the list of supported and unsupported artifacts below
+    # is up to date.
+    supported = ["amis", "gcp"]
+    unsupported = ["aliyun", "azurestack", "digitalocean", "exoscale", "ibmcloud", "powervs", "azure"]
+
+    with open(args.policy, "r") as f:
+        policy = yaml.safe_load(f)
+    validate_policy(stream, policy)
+
+    with open(BUILDFILES['list'], "r") as f:
+        builds_json_data = json.load(f)
+
+    # Prune builds based on the policy
+    for action in ['cloud-uploads', 'images', 'build']:
+        if action not in policy[stream]:
+            continue
+        duration = get_period_in_months(policy[stream][action])
+        ref_date = datetime.datetime.now() - relativedelta(months=int(duration))
+
+        print(f"Pruning resources of type {action} older than {duration} months ({ref_date.date()}) on stream {stream}")
+        # Enumerating in reverse to go from the oldest build to the newest one
+        for index, build in enumerate(reversed(builds_json_data["builds"])):
+            build_id = build["id"]
+            if action in build.get("policy-cleanup", []):
+                print(f"Build {build_id} has already had {action} pruning completed")
+                continue
+            build_date = parse_fcos_version(build_id)
+
+            if build_date >= ref_date:
+                break
+            for arch in build["arches"]:
+                meta_prefix = os.path.join(prefix, f"builds/{build_id}/{arch}/meta.json")
+                meta_json = get_json_from_s3(s3_client, bucket, meta_prefix)
+                # Make sure the meta.json doesn't contain any cloud_platform that is not supported for pruning yet.
+                images = get_supported_images(meta_json, unsupported, supported)
+                current_build = Build(id=build_id, images=images, arch=arch)
+
+                match action:
+                    case "cloud-uploads":
+                        prune_cloud_uploads(current_build, cloud_config, build, builds_json_data, index, args.dry_run)
+                    case "build":
+                        raise NotImplementedError
+                        # print(f"Deleting key {prefix}{build.id} from bucket {bucket}")
+                        # Delete the build's directory in S3
+                        # S3().delete_object(args.bucket, f"{args.prefix}{str(current_build.id)}")
+
+    if not args.dry_run:
+        # Save the updated builds.json to local builds/builds.json
+        save_builds_json(builds_json_data)
+
+    if args.upload_builds_json:
```

Comment:
Again, this is a separate flow. I think let's just hoist this closer to the top of the function. So basically:

```python
def main():
    ...
    if args.upload_builds_json:
        return handle_upload_builds_json(...)
    # continue with regular GC
```